prov/sockets/src/sock_progress.c

```diff
@@ -1520,8 +1520,8 @@ static int sock_pe_process_rx_send(struct sock_pe *pe, struct sock_rx_ctx *rx_ct
 	if (!rx_entry->is_buffered &&
 	    (!(rx_entry->flags & FI_MULTI_RECV) ||
 	     (pe_entry->flags & FI_MULTI_RECV))) {
-		sock_rx_release_entry(rx_entry);
 		fastlock_acquire(&rx_ctx->lock);
+		sock_rx_release_entry(rx_entry);
```

Comment:
Yes, the issue was resetting rx_entry->is_busy at line 1479 before removing from the list at 1497. Another thread was calling sock_rx_release from sock_rx_ctx_cancel() where it was checking is_busy status of rx_entry.

This line here didn't have any impact on the race but I added the release inside a lock to be consistent with other places